moto/ec2/models/spot_requests.py

```diff
@@ -202,6 +203,16 @@ def __init__(
 
         self.launch_specs = []
 
+        self.tags = {}
+        if tag_specifications is not None:
+            tags = convert_tag_spec(tag_specifications)
+            for resource_type in tags:
+                if resource_type != "spot-fleet-request":
+                    raise ValueError(
+                        f"The value for `ResourceType` must be `spot-fleet-request`, but got {resource_type} instead."
+                    )
+            self.tags.update(tags)
+
```

Comment:
Instead of a built-in Python `ValueError` exception, you need to raise an appropriate EC2 custom exception. 

Here is a link to the one you likely want (or should subclass for your specific use-case): https://github.com/getmoto/moto/blob/c9d62e94025fb504881510278150d97d9c4dc002/moto/ec2/exceptions.py#L431